processing/src/main/java/org/apache/druid/query/groupby/GroupingEngine.java

```diff
@@ -788,11 +788,11 @@ private static boolean summaryRowPreconditions(GroupByQuery query)
   private static Iterator<ResultRow> summaryRowIterator(GroupByQuery q)
   {
     List<AggregatorFactory> aggSpec = q.getAggregatorSpecs();
-    Object[] values = new Object[aggSpec.size()];
+    ResultRow resultRow = ResultRow.create(q.getResultRowSizeWithPostAggregators());
     for (int i = 0; i < aggSpec.size(); i++) {
-      values[i] = aggSpec.get(i).factorize(new AllNullColumnSelectorFactory()).get();
+      resultRow.set(i, aggSpec.get(i).factorize(new AllNullColumnSelectorFactory()).get());
     }
```

Comment:
oh, I think you need to actually compute the postaggs... which is why the test is failing, see `mergeResults` for an example, something like
```
  private static Iterator<ResultRow> summaryRowIterator(GroupByQuery q)
  {
    List<AggregatorFactory> aggSpec = q.getAggregatorSpecs();
    ResultRow resultRow = ResultRow.create(q.getResultRowSizeWithPostAggregators());
    for (int i = 0; i < aggSpec.size(); i++) {
      resultRow.set(i, aggSpec.get(i).factorize(new AllNullColumnSelectorFactory()).get());
    }
    Map<String, Object> map = resultRow.toMap(q);
    for (int i = 0; i < q.getPostAggregatorSpecs().size(); i++) {
      final PostAggregator postAggregator = q.getPostAggregatorSpecs().get(i);
      final Object value = postAggregator.compute(map);

      resultRow.set(q.getResultRowPostAggregatorStart() + i, value);
      map.put(postAggregator.getName(), value);
    }
    return Collections.singleton(resultRow).iterator();
  }
```

Reply:
thank you @clintropolis for taking a look and for the suggested fix :)